randovania/resolver/damage_state.py

```diff
@@ -62,9 +60,9 @@ def apply_new_starting_resource_difference(
         """Creates a new state after new resources were added as if they were starting.
         Common difference: collecting a health upgrade fully heals you. But it won't do it here."""
 
-    def resource_requirements_for_satisfying_damage(self, damage: int) -> list[ResourceRequirement]:
+    def resource_requirements_for_satisfying_damage(self, damage: int) -> list[list[ResourceRequirement]]:
```

Comment:
since you're in here, and mypy is complaining: please mark these functions with the `@abstractmethod` decorator from `abc`. if it still complains after doing that, add a `raise NotImplementedError` to each pls